worf/views/list.py

```diff
@@ -125,16 +130,23 @@ def get_processed_queryset(self):
         self._set_base_lookup_kwargs()
         self.set_search_lookup_kwargs()
 
+        query = QueryDict(urlencode(self.lookup_kwargs))
+
         order_by = self.ordering
         if self.request.GET.get("sort"):
             sort = self.parse_sort(self.request.GET.get("sort"))
             if sort.lstrip("-") in self.sort_fields:
                 order_by = [sort]
 
         try:
+            # Not sure this syntax is acceptable, but without it, different filter sets to the same model will fail because django-url-filter uses https://pypi.org/project/cached-property/ to cache the filter set
+            class SelfFilterSet(ModelFilterSet):
+                class Meta(object):
+                    model = self.model
+
```

Comment:
Maybe this is fine, I'd call it `DefaultModelFilterSet` or something and support supplying a custom one via the view:

```py
filter_set = self.filter_set or DefaultModelFilterSet

# ...

filter_set(data=query, queryset=self.get_queryset()).filter()
```

Means that users can supply their own FilterSet of any flavor, and not necessarily a ModelFilterSet.

Reply:
good idea